drizzle-orm/tests/is.test.ts

```diff
@@ -13,4 +13,28 @@ describe.concurrent('is', () => {
 		expect(is(pgExampleTable.a, PgArray)).toBe(true);
 		expect(is(pgExampleTable.a, PgSerial)).toBe(false);
 	});
+
+	test('entityKind prototype chain', ({ expect }) => {
+		class A1 {
+			static readonly [entityKind]: string = 'A';
+		}
+		class A2 {
+			static readonly [entityKind]: string = 'A';
+		}
+
+		class B1 extends A1 {
+			static readonly [entityKind]: string = 'B';
+		}
+
+		class B2 extends B1 {
+			static readonly [entityKind]: string = 'B';
+		}
+
+		const b1 = new B1();
+
+		expect(is(b1, B1)).toBe(true);
+		expect(is(b1, A1)).toBe(true);
+		expect(is(b1, B2)).toBe(true);
+		expect(is(b1, A2)).toBe(true);
```

Comment:
These tests test a case I though was broken, but actually works fine.  Happy to revert, but figured it might be good to have tests for this.

I was not aware that the prototype chain on classes was set up for constructors as well as class instances, because most of my prototype related knowledge predates the introduction of the `class` keyword 😅